src/lua/functions/creatures/npc/npc_functions.cpp

```diff
@@ -524,7 +524,7 @@ int NpcFunctions::luaNpcSellItem(lua_State* L) {
 
 	uint32_t itemsPurchased = 0;
 	uint8_t backpacksPurchased = 0;
-	uint8_t internalCount = it.stackable ? 100 : 1;
+	uint8_t internalCount = it.stackable ? it.stackSize : 1;
```

Comment:
Maybe it is necessary to change to uint16_t?

Reply:
good catch. I ended up not being able to actually get the numbers beyond 255 on the official client, but since I changed it to allow for uint16_t elsewhere it makes sense to be consistent (and perhaps allow for larger stacks with otc)